scripts/stubsabot.py

```diff
@@ -626,14 +683,23 @@ async def determine_action_no_error_handling(
 
     if diff_info is None:
         diff_analysis: DiffAnalysis | None = None
-    else:
-        diff_analysis = await analyze_diff(
+    elif diff_info.host == "github":
+        diff_analysis = await analyze_github_diff(
             github_repo_path=diff_info.repo_path,
             distribution=distribution,
             old_tag=diff_info.old_tag,
             new_tag=diff_info.new_tag,
             session=session,
         )
+    else:
+        assert diff_info.host == "gitlab"
+        diff_analysis = await analyze_gitlab_diff(
+            gitlab_repo_path=diff_info.repo_path,
+            distribution=distribution,
+            old_tag=diff_info.old_tag,
+            new_tag=diff_info.new_tag,
+            session=session,
+        )
```

Comment:
I think we could shorten the if structure in the following way:
```suggestion
    else:
        if diff_info.host == "github":
            analazy_diff = analyze_github_diff
        else:
            analazy_diff = analyze_gitlab_diff
        diff_analysis = await analazy_diff(
            gitlab_repo_path=diff_info.repo_path,
            distribution=distribution,
            old_tag=diff_info.old_tag,
            new_tag=diff_info.new_tag,
            session=session,
        )
```